rust/nix-expr/src/eval_state.rs

```diff
@@ -114,22 +114,27 @@ impl EvalState {
         }
         self.context.check_err()
     }
-    pub fn value_is_thunk(&self, value: &Value) -> bool {
-        let r = unsafe {
-            raw::get_type(self.context.ptr(), value.raw_ptr()) == raw::ValueType_NIX_TYPE_THUNK
-        };
+    pub fn value_type_unforced(&self, value: &Value) -> ValueTypeOrThunk {
+        let r = unsafe { raw::get_type(self.context.ptr(), value.raw_ptr()) };
         self.context.check_err().unwrap();
-        r
-    }
-    pub fn value_type(&self, value: &Value) -> Result<ValueType> {
-        if self.value_is_thunk(value) {
-            self.force(value)?;
+        ValueTypeOrThunk::from_raw(r)
+    }
+    pub fn value_type_forced(&self, value: &Value) -> Result<ValueType> {
```

Comment:
```suggestion
    pub fn value_type(&self, value: &Value) -> Result<ValueType> {
```

We should virtually always use this one anyway, so we can shorten its name.
It's a lot like `builtins.typeOf`; `_unforced` is the weird one that needs a suffix.